testsuite/tests/contrib/lkt_semantic/array_type_expected/test.lkt

```diff
@@ -0,0 +1,25 @@
+val a: Array[Int] = []
+val b: Array[Int] = [1]
+val c: Array[Int] = [1, 2, 3]
+
+@invalid val a = []
```

Comment:
For clarity, I suggest adding comments to explain why invalid declarations are… invalid ^^

Here, for instance:
```
# It is impossible to infer the element type of an empty array literal, without
# the help of type info from the context
@invalid val a = []
```
Also here too, please avoid using the same identifier for two entities in the same scope.

Reply:
Nice, thanks!